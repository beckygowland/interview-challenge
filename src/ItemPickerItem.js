import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { addFoodItem } from './reduxStore';

export class ItemPickerItem extends React.PureComponent {

    addToMenu = () => {
        const { id, name, dietaries, addFoodItem } = this.props;
        addFoodItem({id, name, dietaries})
    }

    render() {
        const { name, dietaries } = this.props;
        return (
            <li className="item" onClick={this.addToMenu}>
                <h2>{name}</h2>
                <p>
                    {dietaries.map((dietary) =>
                        <span key={dietary} className="dietary">{dietary}</span>
                    )}
                </p>
            </li>
        );
    }
}

ItemPickerItem.propTypes = {
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    dietaries: PropTypes.array,
    addFoodItem: PropTypes.func.isRequired,
};

ItemPickerItem.defaultProps = {
    dietaries: [],
};

const mapDispatchToProps = {
    addFoodItem,
};

export default connect(() => ({}), mapDispatchToProps)(ItemPickerItem);