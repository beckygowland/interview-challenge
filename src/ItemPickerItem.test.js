import React from 'react';
import { ItemPickerItem } from './ItemPickerItem';
import sinon from 'sinon';

import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

it('renders', () => {
  const div = shallow(<ItemPickerItem id={1} name="testName"
    dietaries={['a', 'b']} addFoodItem={() => { }} />);
  expect(div).toMatchSnapshot();
  expect(div.find('h2').text()).toBe('testName');
  expect(div.find('span').length).toBe(2);
});

it('onclick calls add food item', () => {
  const onItemClick = sinon.spy();
  const div = shallow(<ItemPickerItem id={1} name="testName"
    addFoodItem={onItemClick} />);
  div.find('li').simulate('click');
  expect(onItemClick.callCount).toBe(1);
  expect(onItemClick.args[0][0].id).toBe(1);
  expect(onItemClick.args[0][0].name).toBe('testName');
  expect(onItemClick.args[0][0].dietaries.length).toBe(0);
});
