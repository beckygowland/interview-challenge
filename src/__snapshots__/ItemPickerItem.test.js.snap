// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemPickerItem
    addFoodItem={[Function]}
    dietaries={
        Array [
            "a",
            "b",
          ]
    }
    id={1}
    name="testName"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          testName
</h2>,
        <p>
          <span
                    className="dietary"
          >
                    a
          </span>
          <span
                    className="dietary"
          >
                    b
          </span>
</p>,
      ],
      "className": "item",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "testName",
        },
        "ref": null,
        "rendered": "testName",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="dietary"
>
              a
</span>,
            <span
              className="dietary"
>
              b
</span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "a",
            "nodeType": "host",
            "props": Object {
              "children": "a",
              "className": "dietary",
            },
            "ref": null,
            "rendered": "a",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "b",
            "nodeType": "host",
            "props": Object {
              "children": "b",
              "className": "dietary",
            },
            "ref": null,
            "rendered": "b",
            "type": "span",
          },
        ],
        "type": "p",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            testName
</h2>,
          <p>
            <span
                        className="dietary"
            >
                        a
            </span>
            <span
                        className="dietary"
            >
                        b
            </span>
</p>,
        ],
        "className": "item",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "testName",
          },
          "ref": null,
          "rendered": "testName",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="dietary"
>
                a
</span>,
              <span
                className="dietary"
>
                b
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "a",
              "nodeType": "host",
              "props": Object {
                "children": "a",
                "className": "dietary",
              },
              "ref": null,
              "rendered": "a",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "b",
              "nodeType": "host",
              "props": Object {
                "children": "b",
                "className": "dietary",
              },
              "ref": null,
              "rendered": "b",
              "type": "span",
            },
          ],
          "type": "p",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
